pkgs/README.md

```diff
@@ -30,12 +30,30 @@ Before adding a new package, please consider the following questions:
-* How realistic is it that it will be used by other people? It's good that nixpkgs caters to various niches, but if it's a niche of 5 people it's probably too small.
+* How realistic is it that it will be used by other people? It's good that nixpkgs caters to various niches, but if it's a niche of 5 people it's probably too small. A good estimate is checking upstream issues and pull requests, or other software repositories.
+* Is the software maintained upstream? Especially packages that are security-critical, rely on fast-moving dependencies or could endanger data integrity should see regular maintenance.
 * Are you willing to maintain the package? You should care enough about the package to be willing to keep it up and running for at least one complete Nixpkgs' release life-cycle.
   * In case you are not able to maintain the package you wrote, you can seek someone to fill that role, effectively adopting the package.
 
 If any of these questions' answer is no, then you should probably not add the package.
 
+Special care has to be taken with security-critical software components.
+* Any package that immediately would need to be tagged with `meta.knownVulnerabilities` is unlikely to be fit for nixpkgs.
+* Any package depending on a known-vulnerable library should be considered carefully.
+* Packages typically used with untrusted data should have a maintained and responsible upstream. For example:
+  * Web engines like Blink, Gecko or Webkit receive frequent updates. Any package which does not keep up-to-date web engines should be considered vulnerable.
+  * Typically publicly exposed web-services working on untrusted input
+  * Archives or rich documents commonly shared by email should be considered as potentially untrusted input, which should be considered for any package working with such files.
+* Applications in the Unix authentication stack such as PAM/DBUS-modules or SUID binaries should be considered carefully, and should have a maintained and responsible upstream.
+* Encryption libraries should have a maintained and responsible upstream.
+* Especially security-critical components that are part of larger packages should be unvendored (=use the nix package as dependency, instead of vendored and pinned sources).
+* A "responsible upstream" includes various aspects, such as:
+  * channels to disclose security concerns
+  * being responsive to security concerns, providing fixes or workarounds
+  * transparent public disclosure of security issues when they are found or fixed
+  * These aspects are sometimes hard to verify, in which case an upstream that is not known to be irresponsible should be considered as responsible.
+* Free software should be built from source where possible. Relying on vendored binary blobs can have devastating effects on security.
```

Comment:
```suggestion
* Free software should be built from source where possible. Relying on vendored binary blobs introduces the risk of supply chain attacks.
```

Reply:
Practically speaking, for every supply chain attack attempt there are hordes of forgotten vulnerable old vendored versions of network-exposed libraries.

Reply:
Fair.

```suggestion
* Source-available software should be built from source where possible. Relying on vendored binary blobs introduces the risk of supply chain attacks and vulnerable outdated dependencies.
```

Also, switched to source-available as it's probably more correct.